jerry-core/vm/vm.h

```diff
@@ -29,12 +29,25 @@
  * @{
  */
 
+/**
+ * Shift for extracting the type of getting input data.
```

Comment:
The description of these four macros is a bit hard to grasp. Would be nice to have them improved. Otherwise, the newly introduced macros look good.